src/app/components/Modals/EOIModal/Step.tsx

```diff
@@ -13,27 +13,34 @@ export const Step = ({
   index,
   state,
   children,
-}: PropsWithChildren<StepProps>) => (
-  <div
-    className={twMerge(
-      "p-4 flex flex-row items-center justify-start gap-3 rounded border border-primary-dark/20 bg-secondary-contrast self-stretch",
-      state !== "processing" && "opacity-25",
-    )}
-  >
-    {state === "completed" ? (
-      <div className="rounded-full bg-primary-light flex h-10 w-10 items-center justify-center">
-        <IoCheckmarkSharp size={24} className="text-secondary-contrast" />
-      </div>
-    ) : (
-      <div className="rounded-full bg-secondary-main flex h-10 w-10 items-center justify-center">
-        <Text variant="body1" className="text-secondary-contrast">
-          {index}
-        </Text>
+}: PropsWithChildren<StepProps>) => {
+  const isCompleted = state === "completed";
+  const isProcessing = state === "processing";
```

Comment:
```suggestion
const ICONS: Record<StepProps["state"], ({ index }: { index: number }) => JSX.Element> = {
  completed: () => <IoCheckmarkSharp size={24} className="text-secondary-contrast" />,
  processing: () => <Loader size={24} className="text-secondary-contrast" />,
  upcoming: ({ index }) => <Text variant="body1" className="text-secondary-contrast">{index}</Text>
}
```
You can move this map from the component